WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPostsParser.swift

```diff
@@ -19,19 +19,25 @@ class BlogDashboardPostsParser {
             return postsDictionary
         }
 
-        if !posts.hasDrafts {
-            // Check for local drafts
-            if let localDraftsCount = numberOfPosts(for: blog, filter: PostListFilter.draftFilter()),
-               localDraftsCount > 0 {
-                posts["draft"] = [0...localDraftsCount].map { _ in [:] }
+        if let localDraftsCount = numberOfPosts(for: blog, filter: PostListFilter.draftFilter()) {
+            if blog.dashboardState.draftsSynced { // If drafts are synced, depend on local data
+                posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
+            }
+            else { // If drafts are not synced, only depend on local data if the cards API returns zero posts
+                if !posts.hasDrafts, localDraftsCount > 0 {
+                    posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
+                }
             }
         }
 
-        if !posts.hasScheduled {
-            // Check for local scheduled
-            if let localScheduledCount = numberOfPosts(for: blog, filter: PostListFilter.scheduledFilter()),
-               localScheduledCount > 0 {
-                posts["scheduled"] = [0...localScheduledCount].map { _ in [:] }
+        if let localScheduledCount = numberOfPosts(for: blog, filter: PostListFilter.scheduledFilter()) {
+            if blog.dashboardState.scheduledSynced { // If scheduled posts are synced, depend on local data
+                posts["scheduled"] = Array(repeatElement([:], count: localScheduledCount))
+            }
+            else { // If scheduled posts are not synced, only depend on local data if the cards API returns zero posts
+                if !posts.hasScheduled, localScheduledCount > 0 {
+                    posts["scheduled"] = Array(repeatElement([:], count: localScheduledCount))
+                }
             }
```

Comment:
Similar to the comment above

```suggestion
            // If scheduled posts are synced OR
            // If scheduled posts are not synced and the cards API returns zero posts
            // depend on local data
            if blog.dashboardState.scheduledSynced || (!posts.hasScheduled && localScheduledCount > 0) {
                posts["scheduled"] = Array(repeatElement([:], count: localScheduledCount))
            }
```

Reply:
Fixed in 652675f769f219409308953b9b46b03eec564d25